src/environmentd/tests/server.rs

```diff
@@ -1035,3 +1036,121 @@ fn test_ws_notifies_for_bad_options() {
         panic!("wrong message!, {notice:?}");
     };
 }
+
+#[derive(Debug, Deserialize)]
+struct HttpResponse<R> {
+    results: Vec<R>,
+}
+
+#[derive(Debug, Deserialize)]
+struct HttpRows {
+    rows: Vec<Vec<String>>,
+    col_names: Vec<String>,
+    notices: Vec<Notice>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Notice {
+    message: String,
+    #[serde(rename = "severity")]
+    _severity: String,
+}
+
+#[test]
+#[cfg_attr(miri, ignore)] // too slow
+fn test_http_passes_options() {
+    let server = util::start_server(util::Config::default()).unwrap();
```

Comment:
Can these tests also be made a table test? They appear to do a lot of the same thing. That would avoid starting one server for each test, which is quite expensive.

Reply:
De-duped the tests here, didn't exactly make it a table test though since the different scenarios seemed to be asserting different enough stuff that trying to unify it into a single function would add too much boiler plate. Happy to revist this if you'd like though :)